goal_src/jak2/engine/gfx/lightning-h.gc

```diff
@@ -88,8 +89,8 @@
    (line           vector-array           :offset-assert 128)
    (meet           vector-array           :offset-assert 132)
    (path           vector-array           :offset-assert 136)
-   (start-color    uint32                 :offset-assert 140)
-   (end-color      uint32                 :offset-assert 144)
+   (start-color    gs-packed-rgba         :offset-assert 140)
+   (end-color      gs-packed-rgba         :offset-assert 144)
```

Comment:
I think this is probably an `rgba`.  As written, this is a reference to a `gs-packed-rgba` structure somewhere else in memory because `gs-packed-rgba` isn't a bitfield type like `rgba`.

Reply:
In here and in `lightning-spec`.

I don't have any evidence, but just based on how their other renderers work, I think it's more likely `rgba`s.

Reply:
Found my justification -- it's because of what i found in `lighting` where `gs-packed-rgba`'s are pulled off the `lightning-control` types and such to be lerp'd:
![image](https://user-images.githubusercontent.com/13153231/195221169-518877fc-ff1a-4194-817d-4d988534428c.png)

Reply:
looks like you're right!
Can you make `lightning-spec` and `lightning-state` use `gs-packed-rgba` - that should get rid of a bunch of casts.

Reply:
When I tried, I ran into static decompilation issues (trying to create a structure from a plain value -- `gs-packed-rgba`s are subtypes of `vector4w`s).  I'd imagine it's similar to however we currently handle decompiling `rgba`s....which im not familiar with yet.

I'll try! :P

Reply:
hm, I just looked a little closer and the `lightning-spec`s are definitely `rgba`...
Maybe `gs-packed-rgba-lerp` actually takes in `rgba`s, and outputs a `gs-packed-rgba`?